internal/label/pullrequests.go

```diff
@@ -9,114 +9,74 @@ import (
 	"github.com/google/go-github/v60/github"
 
 	"github.com/chia-network/github-bot/internal/config"
+	github2 "github.com/chia-network/github-bot/internal/github"
 )
 
 // PullRequests applies internal or community labels to pull requests
 // Internal is determined by checking if the PR author is a member of the specified internalTeam
 func PullRequests(githubClient *github.Client, internalTeam string, cfg config.LabelConfig) error {
-	teamMembers := map[string]bool{}
-
-	teamParts := strings.Split(internalTeam, "/")
-	if len(teamParts) != 2 {
-		return fmt.Errorf("invalid team name - must contain org and team : %s", internalTeam)
-	}
-
-	teamOpts := &github.TeamListTeamMembersOptions{
-		Role: "all",
-		ListOptions: github.ListOptions{
-			Page:    0,
-			PerPage: 100,
-		},
-	}
-
-	for {
-		teamOpts.ListOptions.Page++
-		members, resp, err := githubClient.Teams.ListTeamMembersBySlug(context.TODO(), teamParts[0], teamParts[1], teamOpts)
-		if err != nil {
-			return fmt.Errorf("error getting team %s: %w", internalTeam, err)
-		}
-
-		for _, member := range members {
-			teamMembers[*member.Login] = true
-		}
-
-		if resp.NextPage == 0 {
-			break
-		}
-	}
-
 	for _, fullRepo := range cfg.LabelCheckRepos {
 		log.Println("checking repos")
 		parts := strings.Split(fullRepo.Name, "/")
 		if len(parts) != 2 {
 			return fmt.Errorf("invalid repository name - must contain owner and repository: %s", fullRepo.Name)
 		}
-		opts := &github.PullRequestListOptions{
-			State:     "open",
-			Sort:      "created",
-			Direction: "desc",
-			ListOptions: github.ListOptions{
-				Page:    0,
-				PerPage: 100,
-			},
+		owner := parts[0]
+		repo := parts[1]
+		teamMembers, _ := github2.GetTeamMemList(githubClient, internalTeam)
```

Comment:
This should not be inside of any loops - put it first thing in this function out of the loop. The team member list doesn't change based on what repo we're lookign at, and by doing it in the loop, we're rebuilding this list over and over for no gain (and a cost of a bunch of extra API calls and time)